homeassistant/components/homekit/type_media_players.py

```diff
@@ -289,15 +282,6 @@ def set_mute(self, value):
                   ATTR_MEDIA_VOLUME_MUTED: value}
         self.call_service(DOMAIN, SERVICE_VOLUME_MUTE, params)
 
-    def set_volume(self, value):
```

Comment:
You still need the callback method (`set_volume`). Only the `update_state` section for volume adjustments could be deleted.